core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -67,9 +67,7 @@ object StaticFile {
 
   def fromFile(f: File, start: Long, end: Long, buffsize: Int, req: Option[Request])
                         (implicit es: ExecutorService): Option[Response] = {
-
-    if (start < 0 || end < start || buffsize <= 0)
-      throw new Exception(s"start: $start, end: $end, buffsize: $buffsize")
+    require (start >= 0 && end > start && buffsize > 0, s"start: $start, end: $end, buffsize: $buffsize")
```

Comment:
So it would need to be a `Task[Option[Response]]` then?  With taskiness only to express failure?